tests/eibc_fulfillment_test.go

```diff
@@ -23,6 +24,381 @@ import (
+	isFinalized, err := dymension.WaitUntilRollappHeightIsFinalized(ctx, rollapp1.GetChainID(), rollappHeight, 300)
+	require.NoError(t, err)
+	require.True(t, isFinalized)
+
+	isFinalized, err = dymension.WaitUntilRollappHeightIsFinalized(ctx, rollapp2.GetChainID(), rollapp2Height, 300)
+	require.NoError(t, err)
+	require.True(t, isFinalized)
+
+	expMmBalanceRollappDenom := transferDataRollapp1.Amount
+	balance, err := dymension.GetBalance(ctx, marketMakerAddr, rollappIBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of marketMakerAddr after preconditions:", balance)
+	require.True(t, balance.Equal(expMmBalanceRollappDenom), fmt.Sprintf("Value mismatch. Expected %s, actual %s", expMmBalanceRollappDenom, balance))
+	// end of preconditions
+
+	transferDataRollapp1 = ibc.WalletData{
+		Address: dymensionUserAddr,
+		Denom:   rollapp1.Config().Denom,
+		Amount:  transferAmount,
+	}
+
+	transferDataRollapp2 := ibc.WalletData{
+		Address: dymensionUserAddr,
+		Denom:   rollapp2.Config().Denom,
+		Amount:  transferAmount,
+	}
+
+	// set eIBC specific memo
+	options.Memo = BuildEIbcMemo(eibcFee)
+
+	_, err = rollapp2.SendIBCTransfer(ctx, dymChannel_ra2[0].ChannelID, rollapp2UserAddr, transferDataRollapp2, options)
+	require.NoError(t, err)
+	zeroBalance := math.NewInt(0)
+	balance, err = dymension.GetBalance(ctx, dymensionUserAddr, rollapp2IBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of dymensionUserAddr right after sending eIBC transfer from rollapp 2 to dym hub:", balance)
+	require.True(t, balance.Equal(zeroBalance), fmt.Sprintf("Value mismatch. Expected %s, actual %s", zeroBalance, balance))
+
+	_, err = rollapp1.SendIBCTransfer(ctx, dymChannel_ra1[0].ChannelID, rollappUserAddr, transferDataRollapp1, options)
+	require.NoError(t, err)
+	rollappHeight, err = rollapp1.GetNode().Height(ctx)
+	require.NoError(t, err)
+	balance, err = dymension.GetBalance(ctx, dymensionUserAddr, rollappIBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of dymensionUserAddr right after sending eIBC transfer:", balance)
+	require.True(t, balance.Equal(zeroBalance), fmt.Sprintf("Value mismatch. Expected %s, actual %s", zeroBalance, balance))
+
+	// get eIbc event
+	eibcEvents, err := getEIbcEventsWithinBlockRange(ctx, dymension, 60, false)
+	require.NoError(t, err)
+	for i, eibcEvent := range eibcEvents {
+		fmt.Println(i, "EIBC Event:", eibcEvent)
+	}
+
+	var fulfill_demand_order = false
+	// fulfill demand orders from rollapp 1
+	for _, eibcEvent := range eibcEvents {
+		re := regexp.MustCompile(`^\d+`)
+		if re.ReplaceAllString(eibcEvent.Price, "") == rollappIBCDenom {
+			txhash, err := dymension.FullfillDemandOrder(ctx, eibcEvent.ID, marketMakerAddr)
+			require.NoError(t, err)
+			eibcEvent := getEibcEventFromTx(t, dymension, txhash)
+			if eibcEvent != nil {
+				fmt.Println("After order fulfillment:", eibcEvent)
+			}
+			fulfill_demand_order = true
+		}
+	}
+
+	if !fulfill_demand_order {
```

Comment:
We should make a check for this like `require.Equal(t, true, fulfill_demand_order)` instead of panic here